src/ServiceControl.Monitoring/Http/Diagrams/TimingsAggregator.cs

```diff
@@ -39,6 +44,23 @@ public IEnumerable<MonitoredEndpoint> AggregateIntoLogicalEndpoints()
                     CriticalTime = AggregateTimings(instanceCriticalTime)
                 };
 
+                var queueLength = queueLengths.FirstOrDefault(ql => ql.Key == endpointName);
+                var queueLengthValues = queueLength.Value?.OrderBy(kvp => kvp.Key).ToArray();
+
+                if (queueLengthValues != null)
+                {
+                    var queueLengthMinDate = queueLength.Value.Min(kvp => kvp.Key);
```

Comment:
We just sorted these values above into a collection that we have guaranteed to be non-null. We could just take the first instance as it will be the min.